R/estimate_ascertainment.R

```diff
@@ -0,0 +1,147 @@
+#'
+#' @param type A string, either `"static"` or `"varying"` which determines
+#' whether [estimate_static()] or [estimate_time_varying()] is used to calculate
+#' the resulting ascertainment rate
+#' @param severity_baseline The assumed to be true baseline severity estimate
+#' used in the final ratio to estimate the overall ascertainment rate
+#' @param max_date A string representing a user supplied maximum date, up to
+#' which the time-varying severity estimate will be calculated. Useful in the
+#' case of long time-series, where the user wishes to focus on a specific
+#' time-period
+#'
+#' @return A data.frame containing the MLE estimate and 95% confidence interval
+#' of the corrected severity, named "ascertainment_me" (for the central estimate
+#' ), and "ascertainment_lo" and "ascertainment_hi" for the lower and upper
+#' interval limits.
+#' @export
+#'
+#' @examples
+#' # get data pre-loaded with the package
+#' data("covid_data")
+#' df_covid_uk <- covid_data[covid_data$country == "United Kingdom", ]
+#'
+#' df_covid_uk_subset <- subset(df_covid_uk, date <= "2020-05-31")
+#'
+#' # load epidist object from {epiparameter}
+#' onset_to_death_covid <- epiparameter::epidist_db(
+#'   disease = "COVID-19",
+#'   epi_dist = "onset_to_death",
+#'   author = "Linton_etal"
+#' )
+#'
+#' estimate_ascertainment(
+#'   data = df_covid_uk,
+#'   epidist = onset_to_death_covid,
+#'   type = "varying",
+#'   severity_baseline = 0.014,
+#'   smooth_inputs = TRUE,
+#'   burn_in_value = 7L,
+#'   correct_for_delays = TRUE,
+#'   max_date = "2020-06-30"
+#' )
+#'
+estimate_ascertainment <- function(data,
+                                   epidist = NULL,
+                                   type = c("static", "varying"),
+                                   severity_baseline = 0.014,
+                                   burn_in_value = get_default_burn_in(
+                                     epidist
+                                   ),
+                                   smooth_inputs = FALSE,
+                                   smoothing_window = 1,
+                                   correct_for_delays = FALSE,
+                                   max_date = NULL) {
+  # input checking
+  checkmate::assert_data_frame(data)
+  checkmate::assert_names(
+    colnames(data),
+    must.include = c("date", "cases", "deaths")
+  )
+  checkmate::assert_number(
+    severity_baseline,
+    lower = 0.0, upper = 1.0, finite = TRUE
+  )
+  checkmate::assert_integerish(burn_in_value,
+    lower = 1, len = 1L
+  )
+  checkmate::assert_logical(
+    smooth_inputs,
+    len = 1L, any.missing = FALSE, all.missing = FALSE
+  )
+  checkmate::assert_logical(
+    correct_for_delays,
+    len = 1L, any.missing = FALSE, all.missing = FALSE
+  )
+  checkmate::assert_string(max_date, null.ok = TRUE)
+  if (correct_for_delays) {
+    checkmate::assert_class(epidist, "epidist")
+  }
+
+  # match argument for type
+  type <- match.arg(type, several.ok = FALSE)
```

Comment:
Could you expand on this question, I haven't really understood it? Do you mean that the user might want both the static and the time-varying ascertainment? Would you expect those to be in the same data.frame?

Reply:
I was thinking aloud about the case where a user wants both the static and time-varying results for some kind of comparison. I guess since they don't have the same dimensions, might make sense as a list of results.

Reply:
I see - I think that's easy enough to do manually so let's leave it as it is. Happy to change is there's user demand